importers/data_sources/test/test_data_sources_importer.py

```diff
@@ -0,0 +1,205 @@
+            assert path is not None
+        return importer
+
+
+class Test_data_sources_functions(unittest.TestCase):
+    def test_note(self):
+        cases = [
+            {
+                'input': ['info', 'hi'],
+                'expected': '[importer] ℹ hi\n'
+            },
+            {
+                'input': ['success', 'hello'],
+                'expected': '[importer] ✓ hello\n'
+            },
+            {
+                'input': ['warning', 'uh-oh'],
+                'expected': '[importer] ⚠ uh-oh\n'
+            },
+            {
+                'input': ['error', 'oops'],
+                'expected': '[importer] 🐛 oops\n'
+            },
+            {
+                'input': ['na', 'questionable'],
+                'expected': '[importer] ? questionable\n'
+            },
+        ]
+
+        for case in cases:
+            with patch('sys.stdout', new = io.StringIO()) as mock_stdout:
+                note(*case['input'])
+                self.assertEqual(mock_stdout.getvalue(), case['expected'])
+
+
+
+class Test_data_sources_Importer(unittest.TestCase):
+    # @classmethod
+    # def setUp(cls):
+    #     # import the results file
+    #     # results_file = os.path.join(_TEST_DIR, "djornl", "results.json")
+    #     dir_path = os.path.dirname(os.path.realpath(__file__))
+    #     root_data_path = os.path.join(dir_path, "data")
+    #     with modified_environ(RES_ROOT_DATA_PATH=root_data_path):
+
+    def test_get_config(self):
+        imp = make_importer('standard')
+        self.assertIsNotNone(imp.get_config('API_URL', None))
+        self.assertIsNone(imp.get_config('foo', None))
+        imp = make_importer('standard', warm_config=False)
+        self.assertIsNotNone(imp.get_config('API_URL', None))
+
+    def test_get_config_or_fail(self):
+        imp = make_importer('standard')
+        self.assertIsNotNone(imp.get_config_or_fail('API_URL'))
+        with self.assertRaises(KeyError):
+            imp.get_config_or_fail('foo')
+        imp = make_importer('standard', warm_config=False)
+        self.assertIsNotNone(imp.get_config_or_fail('API_URL'))
+
+    def test_happy_load(self):
+        imp = make_importer('standard')
+        self.assertTrue(imp)
+        result = imp.load_data(dry_run=True)
+        self.assertTrue(result)
+
+    def test_happy_load_default_data_dir(self):
+        """Create importer which uses canonical data"""
+        imp = make_importer('standard', use_env_data=False)
+        result = imp.load_data(dry_run=True)
+        self.assertTrue(result)
+
+    def test_happy_load_bad_dir(self):
+        imp = make_importer('bad_dir')
+        result = imp.load_data(dry_run=True)
+        self.assertFalse(result)
+
+    def test_bad_ns(self):
+        imp = make_importer('bad_ns')
+        result = imp.load_data(dry_run=True)
+        self.assertFalse(result)
+        result = imp.load_data(dry_run=False)
+        self.assertFalse(result)
```

Comment:
Ditto with these tests -- can you add in something that ensures that what you want to be tested is _actually_ being tested?